### PR TITLE
Model bring-up for Qwen/Qwen3-Next-80B-A3B-Instruct

### PR DESCRIPTION
**Pull request by devisettymahidhar608**
### Ticket
https://github.com/tenstorrent/tt-xla/issues/6053

### Problem description
Bring up [`Qwen/Qwen3-Next-80B-A3B-Instruct`](https://huggingface.co/Qwen/Qwen3-Next-80B-A3B-Instruct) as a causal LM. Qwen3-Next is a hybrid decoder: 3 Gated DeltaNet (linear attention) layers + 1 gated full-attention layer, repeated, with a sparse MoE MLP on every layer (512 experts, 10 active, 1 shared). Native in transformers via `Qwen3NextForCausalLM`.

The Gated DeltaNet block runs a depthwise short conv over the fused QKV stream (`Conv1d(8192, 8192, k=4, groups=8192)`). ttnn can only DRAM-slice height/width, and a short prompt (output dim 21) has no spatial split, so the op overflows L1:

```
DRAM Auto slice could not find valid slice configuration.
Tried up to 1 slices for width-slicing on output dimension 21.
Available L1: 1395840 bytes.
```

Patching `nn.Conv1d.forward` is not enough under torch.compile: Dynamo inlines the builtin conv into a single `aten.convolution`.

### What's changed
- Add `qwen_3/causal_lm/moe` loader: tokenizer pad=EOS, `use_cache=False`.
- TP shard spec: low-KV-head full attn (hidden-dim shard), GDN `in_proj_qkvz`/`in_proj_ba` column + replicated `conv1d`, shared-expert column/row; routed experts left to `inject_custom_moe`. Replicate conv activations before the Q/K/V split.
- After load, `slice_depthwise_conv1d_channels(model, chunk=1024)` — same L1 workaround as #912 / #901 / #915.
- `tools/conv_overrides.py`: replace wide depthwise `nn.Conv1d` with `ChannelSlicedDepthwiseConv1d` so Dynamo traces the channel split. Overlaps #901 and #915; rebase whichever lands second.

logs:
[qwencausal_lm.log](https://github.com/user-attachments/files/31965882/qwencausal_lm.log)

### Checklist
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

**Review by sonalibaskaran2499 — APPROVED** [review]

File: qwen_3/causal_lm/moe/pytorch/loader.py [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+                shard_specs[la.in_proj_ba.weight] = ("model", "batch")
+                if hasattr(la, "conv1d"):
+                    shard_specs[la.conv1d.weight] = (None, None, None)
+                shard_specs[la.out_proj.weight] = ("batch", "model")
+                if hasattr(la, "dt_bias"):
+                    shard_specs[la.dt_bias] = ("model",)
+                if hasattr(la, "A_log"):
+                    shard_specs[la.A_log] = ("model",)
+
+        shard_specs[model.lm_head.weight] = ("model", "batch")
```
Comment (kmabeeTT) [inline qwen_3/causal_lm/moe/pytorch/loader.py:229]: Missing sharding on `embed_tokens`? The `qwen_3_5/causal_lm` sibling (and your own #915 Jamba loader) shard `model.model.embed_tokens.weight = ("model","batch")`. Please add it for consistency and re-check PCC.

**Review by kmabeeTT — COMMENTED** [review]
Sharding change

**Review by kmabeeTT — APPROVED** [review]
thx for change - is test passing with high pcc? Can you include link or log in PR Desc if yes.

**Comment by devisettymahidhar608** [comment]
> thx for change - is test passing with high pcc? Can you include link or log in PR Desc if yes.

The model is passing with a 0.99 PCC. I had forgotten to update the PR description, but I’ve updated it now.
